sort.h

```diff
@@ -847,60 +854,324 @@ static void TIM_SORT_RESIZE(TEMP_STORAGE_T *store, const size_t new_size) {
   }
 }
 
-static void TIM_SORT_MERGE(SORT_TYPE *dst, const TIM_SORT_RUN_T *stack, const int stack_curr,
-                           TEMP_STORAGE_T *store) {
-  const size_t A = stack[stack_curr - 2].length;
-  const size_t B = stack[stack_curr - 1].length;
-  const size_t curr = stack[stack_curr - 2].start;
-  SORT_TYPE *storage;
-  size_t i, j, k;
-  TIM_SORT_RESIZE(store, MIN(A, B));
-  storage = store->storage;
 
-  /* left merge */
-  if (A < B) {
-    memcpy(storage, &dst[curr], A * sizeof(SORT_TYPE));
```

Comment:
When fuzzing libxml2, I found that `memcpy` is sometimes be called with a null pointer, maybe because `storage` can be null. I know that it's silly, but `memcpy(NULL, NULL, 0)` is technically undefined behavior and causes errors with UBSan.